src/commons/api/ca.rs

```diff
@@ -1445,21 +1446,27 @@ impl Default for RepoStatus {
     fn default() -> Self {
         RepoStatus {
             last_exchange: None,
+            last_success: None,
             next_exchange_before: Self::now_plus_hours(1),
             published: vec![],
         }
     }
 }
 
 impl RepoStatus {
-    fn next_exchange_before(&self) -> Time {
+    pub fn next_exchange_before(&self) -> Time {
         Time::new(Utc.timestamp(self.next_exchange_before, 0))
```

Comment:
Since `TIme::new(Utc.timestamp(..., 0))` is invoked in multiple places maybe it's worth factoring out to ensure consistency?

Reply:
yeah this is done across the code base.. and in a lot of places (at least for the metrics) we then call .timestamp() again. So perhaps it should just return the timestamps instead.

Reply:
now using new type `Timestamp` and avoiding overhead.